src/crypto/hsm/nxp/PersistentStorageOperationalKeystoreHSM.cpp

```diff
@@ -0,0 +1,279 @@
+                                                                               const Crypto::P256PublicKey & nocPublicKey)
+{
+    ChipLogProgress(Crypto, "SE05x: ActivateOpKeypair for Fabric %02x", fabricIndex);
+
+    VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex) && (fabricIndex == mPendingFabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    // Validate public key being activated matches last generated pending keypair
+    VerifyOrReturnError(mPendingKeypair->Pubkey().Matches(nocPublicKey), CHIP_ERROR_INVALID_PUBLIC_KEY);
+
+    mIsPendingKeypairActive = true;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PersistentStorageOperationalKeystoreHSM::CommitOpKeypairForFabric(FabricIndex fabricIndex)
+{
+    uint8_t i       = 0;
+    uint32_t slotId = mPendingKeypair->GetKeyId() - FABRIC_SE05X_KEYID_START;
+
+    VerifyOrReturnError(slotId != 0, CHIP_ERROR_NO_MEMORY);
+    VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex) && (fabricIndex == mPendingFabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(mIsPendingKeypairActive == true, CHIP_ERROR_INCORRECT_STATE);
+
+    ChipLogProgress(Crypto, "SE05x: CommitOpKeypair for Fabric %02x", fabricIndex);
+
+    while (i++ < MAX_KEYID_SLOTS_FOR_FABRICS)
+    {
+        if (keyidFabIdMapping[i].fabricIndex == fabricIndex)
+        {
+            // Delete the previous keyPair associated with the fabric
+            keyidFabIdMapping[i].isPending = 0;
+            Platform::Delete<Crypto::P256KeypairHSM>(keyidFabIdMapping[i].pkeyPair);
+            keyidFabIdMapping[i].pkeyPair    = NULL;
+            keyidFabIdMapping[i].keyId       = 0;
+            keyidFabIdMapping[i].fabricIndex = 0;
+        }
+    }
+
+    keyidFabIdMapping[slotId].pkeyPair    = mPendingKeypair;
+    keyidFabIdMapping[slotId].keyId       = mPendingKeypair->GetKeyId();
+    keyidFabIdMapping[slotId].fabricIndex = mPendingFabricIndex;
+    keyidFabIdMapping[slotId].isPending   = 0;
+
+    // If we got here, we succeeded and can reset the pending key: next `SignWithOpKeypair` will use the stored key.
+    ResetPendingKey();
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PersistentStorageOperationalKeystoreHSM::RemoveOpKeypairForFabric(FabricIndex fabricIndex)
+{
+    uint8_t i = 0;
+    VerifyOrReturnError(mStorage != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    ChipLogProgress(Crypto, "SE05x: RemoveOpKeypair for Fabric %02x", fabricIndex);
+
+    // Remove pending state if matching
+    if ((mPendingKeypair != nullptr) && (fabricIndex == mPendingFabricIndex))
+    {
+        RevertPendingKeypair();
+    }
+
+    while (i++ < MAX_KEYID_SLOTS_FOR_FABRICS)
```

Comment:
Reads out of bounds on the last iteration.

Reply:
Fixed this.